feat(ui): Dark Mode

Hazer opened the pull request:
## Description
Includes dark mode, auto selects and allows for manually overriding with user preference


### Screenshot
<img width="971" alt="Screenshot 2024-05-02 at 09 30 50" src="https://github.com/LizardByte/Sunshine/assets/4046170/b545167a-997a-4257-a19a-57ef8e3c22c6">
<img width="854" alt="Screenshot 2024-05-02 at 09 31 48" src="https://github.com/LizardByte/Sunshine/assets/4046170/5a4f5fcd-8b28-4d0a-b639-38b41ccc3476">
<img width="826" alt="Screenshot 2024-05-02 at 09 32 04" src="https://github.com/LizardByte/Sunshine/assets/4046170/d84d4857-f290-442f-a2a2-81e91b45b5fd">
<img width="817" alt="Screenshot 2024-05-02 at 09 32 15" src="https://github.com/LizardByte/Sunshine/assets/4046170/ad0d5f67-98ad-45bb-b367-2eb70aed513e">
<img width="263" alt="Screenshot 2024-05-02 at 09 31 23" src="https://github.com/LizardByte/Sunshine/assets/4046170/59f32f70-5572-42aa-9e15-e38cc3c4e90c">
<img width="265" alt="Screenshot 2024-05-02 at 09 31 33" src="https://github.com/LizardByte/Sunshine/assets/4046170/3a3bee02-ba46-4c85-849e-9ebadece3dee">


### Depends on:
- Merging #2491


## Type of Change
- [x] New feature (non-breaking change which adds functionality)
- [ ] Dependency update (updates to dependencies)
- [ ] Documentation update (changes to documentation)
- [ ] Repository update (changes to repository files, e.g. `.github/...`)

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated the in code docstring/documentation-blocks for new or existing methods/components

## Branch Updates
LizardByte requires that branches be up-to-date before merging. This means that after any PR is merged, this branch
must be updated before it can be merged. You must also
[Allow edits from maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [x] I want maintainers to keep my branch updated

Hazer commented:
I didn't ask before, but any considerations regarding colors? Specially with the NavBar colors?

ReenigneArcher commented:
> I didn't ask before, but any considerations regarding colors? Specially with the NavBar colors?

The white(ish) on yellow(ish) is probably going to be difficult to read. I think I'd prefer the same navbar colors for both, e.g. the original. But everywhere else seems to look okay from your screenshots.

Also, I apologize for the delays on getting all these reviewed and in, I'm a bit overwhelmed with this project at the moment, having 50 open PRs. Are you on our discord? If so it would probably easier to communicate about any of these and get them across the goal line.

Hazer commented:
Draft until #2491 is merged.

Hazer commented:
![Screenshot 2024-05-13 at 15 40 31](https://github.com/LizardByte/Sunshine/assets/4046170/c9621d33-581f-4f20-9cef-40e482f26c38)
![Screenshot 2024-05-13 at 15 40 43](https://github.com/LizardByte/Sunshine/assets/4046170/ab9489d6-4c77-48f3-b3c3-ab656b902cbd)

Dark mode navbar colors updated

ReenigneArcher commented on src_assets/common/assets/web/public/assets/locale/en.json:
```diff
@@ -334,7 +334,11 @@
     "password": "Change Password",
     "pin": "Pin",
-    "troubleshoot": "Troubleshooting"
+    "troubleshoot": "Troubleshooting",
+    "toggle_theme": "Toggle theme",
+    "theme_auto": "Auto",
+    "theme_light": "Light",
+    "theme_dark": "Dark"
```
Could you make these alphabetical?

Hazer replied:
```suggestion
    "theme_auto": "Auto",
    "theme_dark": "Dark",
    "theme_light": "Light",
    "toggle_theme": "Toggle theme",
    "troubleshoot": "Troubleshooting"
```

ReenigneArcher commented on src_assets/common/assets/web/Navbar.vue:
```diff
@@ -1,60 +1,93 @@
 <style>
-.nav-link.active {
-    font-weight: 500;
+.navbar-background {
```
Would it make sense to separate out all the styles into the sunshine.css file?

Hazer replied:
I'd rather have each component with it's own style, with time I think it makes things easier, but I was thinking how could I move all the default colors, text sizes and stuff to sunshine.css, kind of creating a design system or theme , but in another PR

ReenigneArcher commented on src_assets/common/assets/web/Navbar.vue:
```diff
@@ -1,60 +1,93 @@
+            <a class="nav-link" href="/config"><i class="fas fa-fw fa-cog"></i> {{ $t('navbar.configuration') }}</a>
+          </li>
+          <li class="nav-item">
+            <a class="nav-link" href="/password"><i class="fas fa-fw fa-user-shield"></i> {{
+                $t('navbar.password')
+              }}</a>
+          </li>
+          <li class="nav-item">
+            <a class="nav-link" href="/troubleshooting"><i class="fas fa-fw fa-info"></i> {{
+                $t('navbar.troubleshoot')
+              }}</a>
```
```suggestion
            <a class="nav-link" href="/password"><i class="fas fa-fw fa-user-shield"></i>
                {{ $t('navbar.password') }}</a>
          </li>
          <li class="nav-item">
            <a class="nav-link" href="/troubleshooting"><i class="fas fa-fw fa-info"></i>
                {{ $t('navbar.troubleshoot') }}</a>
```

Having the i18n opening/closing brackets on separate lines is a bit confusing.

Hazer replied:
CLion reformat got me there, I wasn't expecting this haha

Hazer commented on src_assets/common/assets/web/Navbar.vue:
```diff
@@ -1,60 +1,93 @@
+            <a class="nav-link" href="/config"><i class="fas fa-fw fa-cog"></i> {{ $t('navbar.configuration') }}</a>
+          </li>
+          <li class="nav-item">
+            <a class="nav-link" href="/password"><i class="fas fa-fw fa-user-shield"></i> {{
+                $t('navbar.password')
+              }}</a>
```
```suggestion
            <a class="nav-link" href="/password"><i class="fas fa-fw fa-user-shield"></i> {{ $t('navbar.password') }}</a>
```

Hazer commented on src_assets/common/assets/web/Navbar.vue:
```diff
@@ -1,60 +1,93 @@
+              }}</a>
+          </li>
+          <li class="nav-item">
+            <a class="nav-link" href="/troubleshooting"><i class="fas fa-fw fa-info"></i> {{
+                $t('navbar.troubleshoot')
+              }}</a>
```
```suggestion
            <a class="nav-link" href="/troubleshooting"><i class="fas fa-fw fa-info"></i> {{ $t('navbar.troubleshoot') }}</a>
```

ReenigneArcher commented:
I don't know why, but the toggle drop down doesn't work on the "Applications" tab.

ReenigneArcher commented on src_assets/common/assets/web/public/assets/locale/en.json:
```diff
@@ -334,6 +334,10 @@
+    "theme_auto": "Auto",
+    "theme_dark": "Dark",
+    "theme_light": "Light",
+    "toggle_theme": "Toggle theme",
```
How about just using "Theme"?

Hazer replied:
```suggestion
    "toggle_theme": "Theme",
```

Hazer commented:
> I don't know why, but the toggle drop down doesn't work on the "Applications" tab.

That's odd. I have really no idea why too, there's no obvious reason for it, I'll need to debug later

ReenigneArcher commented:
> > I don't know why, but the toggle drop down doesn't work on the "Applications" tab.
> 
> That's odd. I have really no idea why too, there's no obvious reason for it, I'll need to debug later

@TheElixZammuto any idea what could cause this?

The "button" works on every other navbar page, and there are no errors in the browser console.

Hazer commented:
> The "button" works on every other navbar page, and there are no errors in the browser console.

Fixed. It had to do with bootstrap module bundling being implicitly duplicated. It'll probably be a good idea to revamp the vite configs in the future, for now I just removed the duplication.

ref:
https://github.com/aurelia/skeleton-navigation/issues/894
https://discourse.aurelia.io/t/bootstrap-import-bootstrap-breaks-dropdown-menu-in-navbar/641/9

karasuhebi commented:
@Hazer Thanks for this!
